### PR TITLE
Add algorithmic errors tracking to `qml.Tracker`

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:** Update `qml.Tracker` to track and combine error

**Description of the Change:** Add algorithmic error tracking in the `simulator_tracking` device modifier.

**Benefits:** `tracker` will track algorithmic errors. 

**Possible Drawbacks:** N/A

**Related GitHub Issues:** N/A

### CODE REVIEW

**Comment by obliviateandsurrender** [comment]
[sc-55445]

File: tests/resource/test_error/test_error.py [hunk 1]
```diff
@@ -250,3 +250,21 @@ def test_specs(self):
...
         assert algo_errors["AdditiveError"].error == 0.73 + 0.12
+
+    def test_tracker(self):
+        """Test that tracker are tracking errors as expected."""
+
+        # little hack for stopping device-level decomposition for custom ops
+        def preprocess(execution_config=qml.devices.DefaultExecutionConfig):
+            return qml.transforms.core.TransformProgram(), execution_config
+
+        self.dev.preprocess = preprocess
```
Comment (Jaybsoni) [inline tests/resource/test_error/test_error.py:261]: Why is this needed? Does error tracking fail if the operations get decomposed?

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -176,8 +176,9 @@
   [stim](https://github.com/quantumlib/Stim) `v1.13.0`.
   [(#5409)](https://github.com/PennyLaneAI/pennylane/pull/5409)
 
-* `qml.specs` now returns information regarding algorithmic errors for the qnode as well.
+* `qml.specs` and `qml.Tracker` now returns information regarding algorithmic errors for the qnode as well.
```
Comment (soranjh) [inline doc/releases/changelog-dev.md:179]:
```suggestion
* `qml.specs` and `qml.Tracker` now return information about algorithmic errors for the qnode as well.
```

**Review by soranjh — COMMENTED** [review]
Looks good, left just one minor suggestion. Also, not sure why `preprocess` is needed.

**Review by soranjh — APPROVED** [review]

**Review by Jaybsoni — APPROVED** [review]
💯

**Review by trbromley — COMMENTED** [review]
CC @DSGuala
